src/test/analyserwaveformtest.cpp

```diff
@@ -45,6 +45,49 @@ class AnalyzerWaveformTest : public MixxxTest {
         }
     }
 
+    void assertWaveformReference(
+            ConstWaveformPointer pWaveform,
+            const QString& reference_title) {
+        pWaveform->dump();
+
+        QFile f(QDir::currentPath() + "/src/test/reference_buffers/" + reference_title);
+        bool pass = true;
+        // If the file is not there, we will fail and write out the .actual
+        // reference file.
+        QByteArray actual = pWaveform->toByteArray();
+        QByteArray reference;
+        if (f.open(QFile::ReadOnly)) {
+            reference = f.readAll();
+        }
```

Comment:
It doesn't make sense to continue if the file couldn't be opened.
```suggestion
        ASSERT_TRUE(f.open(QFile::ReadOnly));
        QByteArray reference = f.readAll();
```